techlibs/quicklogic/synth_quicklogic.cc

```diff
@@ -0,0 +1,261 @@
+            run(stringf("hierarchy -check %s", help_mode ? "-top <top>" : top_opt.c_str()));
+        }
+
+        if (check_label("prepare"))  {
+            run("proc");
+            run("flatten");
+            run("opt_expr");
+            run("opt_clean");
+            run("deminout");
+            run("opt");
+        }
+
+        if (check_label("coarse")) {
+            run("opt_expr");
+            run("opt_clean");
+            run("check");
+            run("opt");
+            run("wreduce -keepdc");
+            run("peepopt");
+            run("pmuxtree");
+            run("opt_clean");
+
+            run("alumacc");
+            run("opt");
+            run("fsm");
+            run("opt -fast");
+            run("memory -nomap");
+            run("opt_clean");
+        }
+
+        if (check_label("map_ffram")) {
+            run("opt -fast -mux_undef -undriven -fine");
+            run("memory_map -iattr -attr !ram_block -attr !rom_block -attr logic_block "
+                    "-attr syn_ramstyle=auto -attr syn_ramstyle=registers "
+                    "-attr syn_romstyle=auto -attr syn_romstyle=logic");
+            run("opt -undriven -fine");
+        }
+
+        if (check_label("map_gates")) {
+            if (inferAdder) {
+                run("techmap -map +/techmap.v -map +/quicklogic/" + family + "_arith_map.v");
+            } else {
+                run("techmap");
+            }
+            run("opt -fast");
+            run("opt_expr");
+            run("opt_merge");
+            run("opt_clean");
+            run("opt");
+        }
+
+        if (check_label("map_ffs")) {
+            
+            run("opt_expr -mux_undef");
+            run("simplemap");
+            run("opt_expr");
+            run("opt_merge");
+            run("opt_clean");
+            run("opt");
+        }
+
+        if (check_label("map_luts")) {
+            run("abc -lut 4 ");
+            run("clean");
+            run("opt_lut");
+        }
+
+        if (check_label("map_cells")) {
+
+            std::string techMapArgs;
+            techMapArgs += " -D NO_LUT -map +/quicklogic/" + family + "_lut_map.v";
+            
+            run("techmap" + techMapArgs);
+            run("clean");
+        }
+
+        if (check_label("check")) {
+            run("autoname");
+            run("hierarchy -check");
+            run("stat");
+            run("check -noinit");
+        }
+
+        if (check_label("finalize")) {
+            run("check");
+        }
+
+        if (check_label("edif")) {
+            if (!edif_file.empty())
+                run(stringf("write_edif -nogndvcc -attrprop -pvector par %s %s", this->currmodule.c_str(), edif_file.c_str()));
+        }
+
+        if (check_label("blif")) {
+            if (!blif_file.empty()) {
+                run(stringf("opt_clean -purge"),
+                        "                                 (qlf_k4n8 mode)");
```

Comment:
should this be run here? Why only for blif? It seems `finalize` may be a better place for `opt_clean -purge` (or even earlier stage)

Reply:
Removing, this was present here historically.